app/commands/metric/create.rb

```diff
@@ -6,10 +6,13 @@ class Metric::Create
   def call
     klass = "metrics/#{type}_metric".camelize.constantize
 
-    klass.new(occurred_at:, params:).tap do |metric|
+    klass.new(occurred_at:, country_code:, params: params.except(:remote_ip)).tap do |metric|
       metric.save!
     rescue ActiveRecord::RecordNotUnique
       return klass.find_by!(uniqueness_key: metric.uniqueness_key)
     end
   end
+
+  private
+  def country_code = Geocoder.search(params[:remote_ip]).first&.country_code
```

Comment:
This is where the country code is determined from the remote IP address. This method is always called from within a job, so it should be safe to do so without blocking anything